### PR TITLE
Add regr_intercept and regr_slope Presto aggregate function

### PR DESCRIPTION
**Pull request by liujiayi771**
Add linear regression aggregate functions regr_intercept and regr_slope. https://github.com/facebookincubator/velox/issues/4727

### CODE REVIEW

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp [hunk 1]
```diff
@@ -279,6 +286,97 @@ class CorrIntermediateResult : public CovarIntermediateResult {
...
+struct RegrAccumulator : public CovarAccumulator {
+  double m2X() const {
+    return m2X_;
+  }
+
+  void update(double x, double y) {
+    double oldMeanX = meanX();
+    CovarAccumulator::update(x, y);
+
+    m2X_ += (x - oldMeanX) * (x - meanX());
```
Comment (Yuhta) [inline velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:298]: Should we throw exception if `x == meanX()`?  What is the behavior in Presto Java?

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp [hunk 2]
```diff
@@ -279,6 +286,97 @@ class CorrIntermediateResult : public CovarIntermediateResult {
...
+      double meanYOther,
+      double c2Other,
+      double m2XOther) {
+    if (countOther == 0) {
+      return;
+    }
+
+    m2X_ += m2XOther +
+        count() * countOther * std::pow(meanX() - meanXOther, 2) /
+            (double)(count() + countOther);
```
Comment (Yuhta) [inline velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:313]:
```suggestion
    m2X_ += m2XOther +
        count() / (count() + countOther) * countOther * std::pow(meanX() - meanXOther, 2);
```

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp (hunk 1 above)
Comment (liujiayi771) [reply velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:298]: Presto Java not `check x == meanX()`. https://github.com/prestodb/presto/blob/1edcf2dc2a9ed67ef7a3ce8962601c8246206294/presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationUtils.java#L149

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp [hunk 3]
```diff
@@ -279,6 +286,97 @@ class CorrIntermediateResult : public CovarIntermediateResult {
...
+    CovarAccumulator::merge(countOther, meanXOther, meanYOther, c2Other);
+  }
+
+ private:
+  double m2X_;
```
[the remark this answers is not in the document]
Comment (liujiayi771) [reply velox/functions/prestosql/aggregates/CovarianceAggregates.cpp]: Also fixed double initialization in CorrAccumulator.

**Comment by liujiayi771** [comment]
Ubuntu Benchmark failed in Install dependencies.
```
E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
```

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp (hunk 1 above)
Comment (Yuhta) [reply velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:298]: Oh never mind it should not matter in this case

**Comment by liujiayi771** [comment]
@Yuhta Is there anything else that needs to be modified?

**Pull request closed without merging**